app/main.py

```diff
@@ -2,0 +1,33 @@
+from __future__ import annotations
+
+
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100,
+                 hidden: bool = False) -> None:
+        self.health = health
+        self.name = name
+        self.hidden = hidden
+        Animal.alive.append(self)
```

Comment:
Appending `self` to `Animal.alive` in the constructor is correct for tracking instances. However, make sure that the `die` method is called appropriately to remove instances from this list when necessary.